server/src/main/java/com/lunchtime/service/impl/PersonServiceImpl.java

```diff
@@ -1,24 +1,53 @@
 package com.lunchtime.service.impl;
 
+import com.lunchtime.mapper.PersonMapper;
 import com.lunchtime.models.Person;
+import com.lunchtime.service.dto.PersonDto;
+import com.lunchtime.service.dto.PersonPassDto;
 import com.lunchtime.repository.PersonRepository;
 import com.lunchtime.service.PersonService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 
 @Service
+@RequiredArgsConstructor
 public class PersonServiceImpl implements PersonService {
     private final PersonRepository personRepository;
-
-    public PersonServiceImpl(PersonRepository personRepository) {
-        this.personRepository = personRepository;
-    }
+    private final PersonMapper personMapper;
 
     public Person savePerson(Person person) {
         return personRepository.save(person);
     }
 
+    public PersonDto updatePerson(PersonDto personDto) {
+        Person person = personMapper.fromDtoToPerson(personDto);
+        personRepository.save(person);
+        return personDto;
+    }
+
+    public PersonPassDto updatePassword(PersonPassDto personPassDto) throws Exception {
+        Optional<Person> result = personRepository.findById(personPassDto.getId());
+        if (result.isPresent()) {
+            Person person = result.get();
+            if (personPassDto.getOldPassword().equals(person.getPassword())) {
+                person.setPassword(personPassDto.getPassword());
+                person.setName(personPassDto.getName());
+                person.setPhoneNumber(personPassDto.getPhoneNumber());
+                personRepository.save(person);
+                return personPassDto;
+            }
+            throw new Exception();
```

Comment:
Why do we need to throw here an Exception?

Reply:
I throw this exception if we get wrong old password from front-end.

Reply:
May be you forgot `else if` before throwing this Exception?

Reply:
This is not necessary, because we have `return`.